### PR TITLE
Document `hf_transfer` more prominently

### PR DESCRIPTION
**Pull request by Wauplin**
Discussed in https://github.com/huggingface/huggingface_hub/issues/1652 and [slack](https://huggingface.slack.com/archives/C03V11RNS7P/p1694186667458409) (private) cc @Narsil @osanseviero 

This PR adds more info about `hf_transfer` in the docs and promote its usage in the CLI upload/download commands. Still, it is not set by default -and is not planned to be- because of its inherent limitations (progress bars, error handling, resumable download,...). The goal is to make this feature more known while being optional. About the CLI commands if users don't want any log message (including this one), they can set `--quiet` so it should "pollute" too much the output.

I also created an alias so that https://huggingface.co/docs/huggingface_hub/hf_transfer redirects to the corresponding section in the docs.

### CODE REVIEW

File: docs/source/en/package_reference/environment_variables.md [hunk 1]
```diff
@@ -136,17 +136,15 @@ You can set `HF_HUB_DISABLE_TELEMETRY=1` as environment variable to globally dis
...
 ### HF_HUB_ENABLE_HF_TRANSFER
 
-Set to `True` to download files from the Hub using `hf_transfer`. It's a Rust-based package
-that enables faster download (up to x2 speed-up). Be aware that this is still experimental
-so it might cause issues in your workflow. In particular, it does not support features such
-as progress bars, resume download, proxies or error handling.
+Set to `True` for faster uploads and downloads from the Hub using `hf_transfer`.
 
-**Note:** `hf_transfer` has to be installed separately [from Pypi](https://pypi.org/project/hf-transfer/).
+By default, `huggingface_hub` uses the Python-based `requests.get` and `requests.post` functions. Although these are reliable and versatile, they may not be the most efficient choice for machines with high bandwidth. `hf_transfer`, a Rust-based package, was developed to maximize the bandwidth used by dividing large files into smaller parts and transferring them simultaneously using multiple threads. This approach can potentially double the transfer speed. To use `hf_transfer`, you need to install it separately [from Pypi](https://pypi.org/project/hf-transfer/) and set `HF_HUB_ENABLE_HF_TRANSFER=1` as environment variable.
```
Comment (stevhliu) [inline docs/source/en/package_reference/environment_variables.md:141]:
```suggestion
By default, `huggingface_hub` uses the Python-based `requests.get` and `requests.post` functions. Although these are reliable and versatile, they may not be the most efficient choice for machines with high bandwidth. [`hf_transfer`](https://github.com/huggingface/hf_transfer) is a Rust-based package developed to maximize the bandwidth used by dividing large files into smaller parts and transferring them simultaneously using multiple threads. This approach can potentially double the transfer speed. To use `hf_transfer`, you need to install it separately [from PyPI](https://pypi.org/project/hf-transfer/) and set `HF_HUB_ENABLE_HF_TRANSFER=1` as an environment variable.
```

File: docs/source/en/package_reference/environment_variables.md [hunk 2]
```diff
@@ -136,17 +136,15 @@ You can set `HF_HUB_DISABLE_TELEMETRY=1` as environment variable to globally dis
...
+
+Please note that using `hf_transfer` comes with certain limitations. Since it is not purely Python-based, debugging errors may be challenging. Additionally, `hf_transfer` lacks several user-friendly features such as progress bars, resumable downloads and proxies. These omissions are intentional to maintain the simplicity and speed of the Rust logic. Consequently, `hf_transfer` is not enabled by default in `huggingface_hub`.
 
 ## From external tools
 
-Some environment variables are not specific to `huggingface_hub` but still taken into account
-when they are set.
+Some environment variables are not specific to `huggingface_hub` but still taken into account when they are set.
```
Comment (stevhliu) [inline docs/source/en/package_reference/environment_variables.md:147]:
```suggestion
Some environment variables are not specific to `huggingface_hub` but are still taken into account when they are set.
```

**Review by stevhliu — APPROVED** [review]
Super cool and good job being transparent about the lack of user-friendly features 😄

**Comment by Wauplin** [comment]
Thanks again for the great feedback @stevhliu! And yes always good to lower users expectations :smile:

**Review by LysandreJik — APPROVED** [review]
Nice, looks good to me!

Thanks @Wauplin

**Comment by Wauplin** [comment]
Thanks for the review, I'll merge :)

**Pull request merged**